packages/migrations/src/actions/2024.11.12T00-00-00.supertokens-9.2.ts

```diff
@@ -0,0 +1,15 @@
+import { type MigrationExecutor } from '../pg-migrator';
+
+export default {
+  name: '2024.11.12T00-00-00.supertokens-9.2.ts',
+  run: ({ sql }) => sql`
+    DO $$
+    BEGIN
+      IF (SELECT to_regclass('supertokens_user_last_active') IS NOT null)
```

Comment:
Supertoken database migrations are a bit messy.

We have two scenarios:
1. Upgrading existing Cloud Hive Console / Self-Hosted Hive Console
2. Migrating a completely fresh Hive Console instance

In scenario 1, the `supertokens_user_last_active` table is already guaranteed to exist, and the `BEGIN IF THEN` will be run.

In scenario 2, the supertokens service will race with the migrations runner.

If the supertokens service wins, it will create/seed the complete supertokens database schema, and later on when the migration service runs these migrations are simply skipped (`CREATE INDEX IF NOT EXISTS`).

If the migrations runner wins, this block will be skipped as `supertokens_user_last_active` does not exist like in scenario 1, and we pass responsibility to creating ALL the tables to the supertokens service.